### PR TITLE
fix: missing command

### LINKED ISSUE #101: $cluster.name is empty, PowerManagement-ManagementDomain.ps1 errors out
### Code of Conduct

- [X] I have read and agree to the [Code of Conduct](https://github.com/vmware/powershell-module-for-cloud-foundation-power-management/blob/main/CODE_OF_CONDUCT.md).
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### VMware Cloud Foundation

5.1

### PowerShell Version

7.4.1

### PowerCLI Version

13.2

### Module Version

1.4.1

### PowerVCF Version

2.4.1

### Guest Operating System

windows server 2019

### Environment Details

_No response_

### Description

line: 395 
the line failed due to $cluster.name is not assigned anywhere.  this non assignment affected every other cmdlet calls afterwards.  

### Error or Debug Output

 [04-02-2024_14:06:39] ERROR  ERROR at Script Line 395
Write-PowerManagementLogMessage: C:\Users\kteng\Documents\PowerShell\Modules\VMware.CloudFoundation.PowerManagement\1.4.1.1009\SampleScripts\PowerManagement-ManagementDomain.ps1:101
Line |
 101 |      Write-PowerManagementLogMessage -Type ERROR -Message " ERROR at S …
     |      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     |  ERROR at Script Line 395
PS C:\Users\kteng\Documents\PowerShell\Modules\VMware.CloudFoundation.PowerManagement\1.4.1.1009\SampleScripts>

### Expected Behavior

the cmdlet call goes through and return correct evaluation.

### Actual Behavior

cmdlet error out due to unknown parameter input.  

### Steps to Reproduce

install modules
run sample script
$sddcManagerFqdn = "sfo-vcf01.sfo.rainpole.io"
$sddcManagerUser = "administrator@vsphere.local"
$sddcManagerPass = "VMw@re1!"

./PowerManagement-ManagementDomain.ps1 -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -shutdown -shutdownCustomerVm

### Log Fragments and Files

_No response_

### Screenshots

_No response_

### References

_No response_

### PR DESCRIPTION
**Pull request by burnsjared0415**
### Summary
Command `$cluster = Get-VCFCluster | Where-Object { $_.id -eq ($workloadDomain.clusters.id) }` was removed on accident, also updated a couple variables

### Type

- [x] Bugfix
- [ ] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [x] Yes, there are breaking changes.
- [ ] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [ ] Documentation has been added or updated.

### Issue References
Closes #101

### CODE REVIEW

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 1]
```diff
@@ -451,13 +454,13 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
             $nsxtMgrfqdn = $nsxtCluster.vipFqdn
             $nsxMgrVIP = New-Object -TypeName PSCustomObject
             $nsxMgrVIP | Add-Member -Type NoteProperty -Name adminUser -Value (Get-VCFCredential | Where-Object ({ $_.resource.resourceName -eq $nsxtMgrfqdn -and $_.credentialType -eq "API" })).username
-            $Pass = (Get-VCFCredential | Where-Object ({ $_.resource.resourceName -eq $nsxtMgrfqdn -and $_.credentialType -eq "API" })).password
-            if ($Pass) {
-                $Pass_encrypted = $Pass | ConvertTo-SecureString -AsPlainText -Force | ConvertFrom-SecureString
+            $nsxmgrPass = (Get-VCFCredential | Where-Object ({ $_.resource.resourceName -eq $nsxtMgrfqdn -and $_.credentialType -eq "API" })).password
```
Comment (tenthirtyam) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:457]:
```suggestion
            $nsxMgrPass = (Get-VCFCredential | Where-Object ({ $_.resource.resourceName -eq $nsxtMgrfqdn -and $_.credentialType -eq "API" })).password
```

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 2]
```diff
@@ -451,13 +454,13 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
...
+            if ($nsxmgrPass) {
```
Comment (tenthirtyam) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:458]:
```suggestion
            if ($nsxMgrPass) {
```

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 3]
```diff
@@ -451,13 +454,13 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
...
+                $Pass_encrypted = $nsxmgrPass | ConvertTo-SecureString -AsPlainText -Force | ConvertFrom-SecureString
```
Comment (tenthirtyam) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:459]:
```suggestion
                $nsxMgrPassEncrypted = $nsxMgrPass | ConvertTo-SecureString -AsPlainText -Force | ConvertFrom-SecureString
```

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 4]
```diff
@@ -451,13 +454,13 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
...
             } else {
                 $Pass_encrypted = $null
```
Comment (tenthirtyam) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:461]: $nsxMgrPassEncrypted

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 5]
```diff
@@ -194,9 +195,9 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
                 $mgmtClusterIds = (Get-VCFWorkloadDomain | Select-Object Type -ExpandProperty clusters | Where-Object { $_.type -eq "MANAGEMENT" }).id
                     foreach ($clusterid in $mgmtClusterIds) {
                         $clusterid = (Get-VCFCluster | Select-Object name, id, isdefault | Where-Object { $_.id -eq $clusterid })
-                        $clustername = $Clusterid.name
+                        $clustername_extra = $Clusterid.name
```
Comment (tenthirtyam) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:198]: Was there a variable conflict?

**Review by tenthirtyam — CHANGES_REQUESTED** [review]

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 6]
```diff
@@ -194,9 +195,9 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
...
                         if (!$isDefault) {
-                            $answer = Read-Host -Prompt "Shutdown cluster $clustername. Do you want to continue? Y/N"
+                            $answer = Read-Host -Prompt "Shutdown cluster $clustername_extra. Do you want to continue? Y/N"
                             if ($answer -Match "N") {
                                 Write-PowerManagementLogMessage -Type WARNING "Cancelling shutdown of $clustername. Exiting..."
```
Comment (garlicNova) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:202]:
```suggestion
                                Write-PowerManagementLogMessage -Type WARNING "Cancelling shutdown of $clustername_extra. Exiting..."
```

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 7]
```diff
@@ -211,7 +212,7 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
                                         # Set DRS Automation Level to Manual in the Management Domain
                                         Set-DrsAutomationLevel -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername -level Manual
```
Comment (garlicNova) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:213]:
```suggestion
                                        Set-DrsAutomationLevel -server $vcfVcenterDetails.fqdn -user $vcfVcenterDetails.ssoAdmin -pass $vcfVcenterDetails.ssoAdminPass -cluster $clustername_extra -level Manual
```

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 8]
```diff
@@ -363,15 +364,17 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
                     Write-PowerManagementLogMessage -Type INFO -Message "A single cluster exists in the management domain."
             }
     
-
+            $cluster = Get-VCFCluster | Where-Object { $_.id -eq ($workloadDomain.clusters.id) }
```
Comment (garlicNova) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:367]: $cluster = Get-VCFCluster | Where-Object { $_.id -eq ($workloadDomain.clusters.id) }

should be within the else{} for when there's a single cluster exists in the management domain.

**Review by garlicNova — COMMENTED** [review]
There seemed to be validation issue with this section.  by changing $clustername to $clustername_extra functional breaks for prior cmdlet calls at line 205, 228, 250, 251, 257, 267 and 330.

**Comment by garlicNova** [comment]
I am unable to validate the commit due to other issues as well.  
line: 196 foreach ($clusterid in $mgmtClusterIds) when the cluster in which sfo-m01-vc01 vm exists is higher than other cluster in this list, the function will shutdown sfo-m01-vc01 vm before other clusters in the domain shutdown down.  
line 349: if i sort the cluster list to have the main cluster be the last one on the list, the script ends at line 349 and the startup json is not created, therefore i am unable to powerup the environment through the script.  
Please make the above changes and re-run the module E2E based on step 6 in https://vmware.github.io/powershell-module-for-vmware-cloud-foundation-power-management/documentation/#usage-examples

**Comment by garlicNova** [comment]
> line: 196 foreach ($clusterid in $mgmtClusterIds) when the cluster in which sfo-m01-vc01 vm exists is higher than other cluster in this list, the function will shutdown sfo-m01-vc01 vm before other clusters in the domain shutdown down. i am not sure understand this comment. I am really not sure what the issues are, i can't replicate these issues

ah i found the issue, line 199

File: SampleScripts/PowerManagement-ManagementDomain.ps1 [hunk 9]
```diff
@@ -194,11 +195,11 @@ if ($PsBoundParameters.ContainsKey("shutdown") -or $PsBoundParameters.ContainsKe
                 $mgmtClusterIds = (Get-VCFWorkloadDomain | Select-Object Type -ExpandProperty clusters | Where-Object { $_.type -eq "MANAGEMENT" }).id
                     foreach ($clusterid in $mgmtClusterIds) {
                         $clusterid = (Get-VCFCluster | Select-Object name, id, isdefault | Where-Object { $_.id -eq $clusterid })
-                        $clustername = $Clusterid.name
+                        $clustername_extra = $Clusterid.name
                         if (!$isDefault) {
```
Comment (garlicNova) [inline SampleScripts/PowerManagement-ManagementDomain.ps1:199]:
```suggestion
                        if (!$clusterid.isDefault) {
```

**Comment by tenthirtyam** [comment]
Merging this one after discussions with @joisika. He will review the current state and initiate some in-depth testing for the next release.

**Pull request merged**